### PR TITLE
chore: specify required/recommended guidelines

### PR DESCRIPTION
**Pull request by toddbaert**
* specifies which points are required, as requested by CNCF
* alphabetized within required/recommended

---

Probably easiest to review this in preview mode: https://github.com/open-feature/.github/pull/54/files?short_path=eca12c0#diff-eca12c0a30e25b4b46522ebf89465a03ba72a03f540796c979137931d8f92055

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -18,22 +18,25 @@ Project governance is documented [here](https://github.com/open-feature/communit
...
+| dependency auto-updates         | [Renovate][renovate]\*\*, [Dependabot][dependabot]                                                                                        | required                                                    |
+| image signing                   | language-specific tools, [cosign][cosign]                                                                                                 | required                                                    |
+| semantic versioning, changelogs | [Semantic Versioning][semantic-versioning], [Conventional Commits][conventional-commits], [Release Please][release-please]\*\*\*          | required                                                    |
+| unit, integration testing       | github actions (with permissions applying principle of least privilege), language-specific tools (JUnit, Jest, etc), [Cucumber][cucumber] | required, with coverage metrics up to maintainer discretion |
+| artifact signing                | language-specific tools, [cosign][cosign]                                                                                                 | recommended, where supported by language-specific tools     |
```
Comment (toddbaert) [inline CONTRIBUTING.md:35]: I'd like to make this one required, but it's difficult, and simply not supported by a lot of package managers.

**Review by kinyoklion — COMMENTED** [review]
I think it looks good. I have a couple thoughts, but they aren't strong.

When I read `image signing` that was ambiguous to me. It seems to be `container image signing`. Sometimes with artifacts image can take a more wide meaning.

The second is the other signing one. Something like `npm publish --provenance` would probably meet  a `artifact signing` requirement, as part of it involves signing. I wonder though is provenance a more general requirement than specifically signing. Probably just a philosophical point.

**Comment by toddbaert** [comment]
> When I read `image signing` that was ambiguous to me. It seems to be `container image signing`. Sometimes with artifacts image can take a more wide meaning.

Good point.

> The second is the other signing one. Something like `npm publish --provenance` would probably meet a `signing` requirement, as part of it involves signing. I wonder though is provenance a more general requirement than specifically signing. Probably just a philosophical point.

I would say that provenance is _meaningfully_ different than signing (though more general). I think it's worth noting. TBH eventually I'd like to have [SLSA](https://slsa.dev/) provenance/attestations included where reasonable. I will try to be a bit more precise with this requirement.

**Comment by toddbaert** [comment]
@kinyoklion I've merged the signing requirements, and mentioned "container images". I've also added a "provenance" item. I wanted to say something like `build provenance", but I know SLSA intends to add more forms of provenance associated with things other than builds, so I'm not sure I can be more specific.

**Review by kinyoklion — APPROVED** [review]

**Review by DavidPHirsch — APPROVED** [review]

**Pull request merged**